src/core/execute.c

```diff
@@ -4143,6 +4164,17 @@ void exec_runtime_destroy(ExecRuntime *rt) {
         safe_close_pair(rt->netns_storage_socket);
 }
 
+void exec_params_destroy(ExecParameters *p) {
+        int i;
+
+        if (!p)
+                return;
+
+        for (i = 0; i < _EXEC_DIRECTORY_MAX; i++)
+                if (p->prefix[i])
+                        p->prefix[i] = mfree(p->prefix[i]);
```

Comment:
free/mfree happily take NULL points, and become NOPs then. You hence can drop the if check here, and just directly invoke mfree...